### PR TITLE
Add branch-specific premerge jenkinsfile [skip ci]

### PR DESCRIPTION
**Pull request by pxLi**
Signed-off-by: Peixin Li <pxli@nyu.edu>

To support run premerge w/ branch-specific jenkinsfile

Generally copied from internal pre-merge jenkinsfile, but modified params to support being triggered by upstream router job

### CODE REVIEW

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Review by zhanga5 — APPROVED** [review]
LGTM

**Comment by pxLi** [comment]
Please allow me to merge this one, thanks!

I will merge it after all internal and github action setup got updated~

File: ci/Jenkinsfile.premerge [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        ansiColor('xterm')
+        buildDiscarder(logRotator(numToKeepStr: '50'))
+        skipDefaultCheckout true
+        timeout(time: 12, unit: 'HOURS')
+    }
+
+    parameters {
+        string(name: 'PARALLEL_LEVEL', defaultValue: '18',
+            description: 'Parallel build cudf cpp with -DCPP_PARALLEL_LEVEL')
+        string(name: 'REF', defaultValue: '',
```
Comment (NvTimLiu) [inline ci/Jenkinsfile.premerge:60]: Did not found where this parameter is used?

Comment (pxLi) [reply ci/Jenkinsfile.premerge:60]: > Generally copied from internal pre-merge jenkinsfile, but modified params to support being triggered by upstream router job

Comment (pxLi) [reply ci/Jenkinsfile.premerge:60]: So it will be used in pipeline configure to fetch jenkinsfile based on input $REF

Comment (NvTimLiu) [reply ci/Jenkinsfile.premerge:60]: Got it

**Pull request merged**